projects/Yay!/index.js

```diff
@@ -0,0 +1,72 @@
+const yayStoneAddress = '0xe86142af1321eaac4270422081c1EdA31eEcFf0c'
+const yayAgETHAddress = '0x0341d2c2CE65B62aF8887E905245B8CfEA2a3b97'
+
+const yayStoneOFTAdapterAddress = '0x4b18d95B3CA275AdaD67F1dC81c0FE0D1FB58d59'
+const yayAgEthOFTAdapterAddress = '0xA8ab83461e443f4a24613C1EFB7D1C653F95dD4C'
+
+const eth = async (api) => {
+  // original balances
+  const balances = await api.sumTokens({
+    tokensAndOwners: [
+      ['0x7122985656e38bdc0302db86685bb972b145bd3c', yayStoneAddress],
+      ['0xe1b4d34e8754600962cd944b535180bd758e6c2e', yayAgETHAddress],
+    ],
+  })
+
+  // bridged balances
+  const bridgedYayStoneBals = await api.call({
+    target: yayStoneAddress,
+    params: [yayStoneOFTAdapterAddress],
+    abi: 'erc20:balanceOf',
+  })
+
+  const bridgedYayAgEthBals = await api.call({
+    target: yayAgETHAddress,
+    params: [yayAgEthOFTAdapterAddress],
+    abi: 'erc20:balanceOf',
+  })
+
+  // deduct bridged balances
+  balances["ethereum:0x7122985656e38bdc0302db86685bb972b145bd3c"] = balances["ethereum:0x7122985656e38bdc0302db86685bb972b145bd3c"] - bridgedYayStoneBals
+  balances["ethereum:0xe1b4d34e8754600962cd944b535180bd758e6c2e"] = balances["ethereum:0xe1b4d34e8754600962cd944b535180bd758e6c2e"] - bridgedYayAgEthBals
+
+  return balances;
+}
+
+const soneium = async (api) => {
```

Comment:
for a bridge, we count tokens locked in the source chain, can you share the chain & contract where astar is locked? as I have mentioned previously, we cant count the yay tokens